execution/src/transaction.rs

```diff
@@ -199,30 +344,22 @@ impl NormalizedEthTransaction {
     }
 }
 
-impl TryFrom<TxEnvelope> for NormalizedEthTransaction {
-    type Error = Error;
-
-    fn try_from(tx: TxEnvelope) -> Result<Self, Self::Error> {
-        Ok(match tx {
-            TxEnvelope::Eip1559(tx) => tx.try_into()?,
-            TxEnvelope::Eip2930(tx) => tx.try_into()?,
-            TxEnvelope::Legacy(tx) => tx.try_into()?,
-            TxEnvelope::Eip4844(_) | TxEnvelope::Eip7702(_) => {
-                Err(InvalidTransactionCause::UnsupportedType)?
-            }
-        })
-    }
-}
-
 impl TryFrom<Signed<TxEip1559>> for NormalizedEthTransaction {
     type Error = Error;
 
     fn try_from(value: Signed<TxEip1559>) -> Result<Self, Self::Error> {
         let address = value.recover_signer()?;
-        let tx = value.strip_signature();
+
+        let mut encoded = Vec::new();
+        value.rlp_encode(&mut encoded);
+        let l1_gas_fee_input = encoded.as_slice().into();
```

Comment:
I don't really like needing to completely re-encode the transaction to get this information. Can the flow be modified somehow so that instead of using conversion traits we have constructor methods that take in the gas information as an argument?

Reply:
Made it into a builder-like method in e6e5b3e.